pkg/extension/sumologicextension/extension.go

```diff
@@ -268,6 +283,61 @@ func (pm *sumologicExtension) register(ctx context.Context, collectorName string
 	return nil
 }
 
+func (pm *sumologicExtension) heartbeat(ctx context.Context) {
```

Comment:
I'd suggest to refactor this a bit:

* make this `heartbeatLoop()` or something similar
* add a func that **only** sends a heartbeat and use it in `default:` case in the loop
* remove context param from this func's param list, it's coming from `Start`'s param list and will outlive `Start`